### ISSUE TITLE
The validator should not report "Row ID should start with a lowercase letter" errors

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord/tc-create-app#704]
v1.1.0
Validation file: [https://app.zenhub.com/files/191973535/bbaf46a1-adf5-4a47-b0b4-9c6c2923aef4/download](https://app.zenhub.com/files/191973535/bbaf46a1-adf5-4a47-b0b4-9c6c2923aef4/download)

I don't think there is any way for the user to change the source ID field. If that is so, it seems undesirable to suggest that they modify the target ID field. Would the app even know how to handle it if the ID fields did not match?

![image.png](https://images.zenhubusercontent.com/56e1a9316caf56c9d38d8df5/be1a89f9-ac87-4ac2-8688-c5165b1f0fec)

**Comment by mandolyte** [comment]
@RobH123 thoughts?

These look suspiciously like they were entered by hand

**Comment by RobH123** [comment]
Yes, it's common for current users to edit TSV files with software like spreadsheet editors or code editors, and so they make up IDs and are not always aware of the spec for that field @mandolyte .

**Comment by birchamp** [comment]
@mandolyte can we find out if tC Create can create ID's starting with a number?

**Comment by mandolyte** [comment]
@klappy I find the code obscure for generating the row id. I suppose because the code is supposed be generic it is using an indirect method of determining when to create one?

My best guess is that this is the function that is doing it (located in datatable.js):
```js
const randomId = ({ length }) => {
  const number = Math.random(); // 0.9394456857981651

  // number.toString(36); // '0.xtis06h6'
  if (length > 9) {
    length = 9;
  }

  const id = number.toString(36).substr(2, length); // 'xtis06h6'
  return id;
};

console.log(randomId({length: 4}))
```
I put the above with console logger and found that it does occasionally create an id with a leading digit.

Is the above the right code? The logic behind when it is called is hard to figure out. I'd happy to add some comments if you could explain why it is doing what it does.

@birchamp

**Comment by birchamp** [comment]
@mandolyte calculate the first character by running  number.toString(26) so it's always alphabetical.

**Comment by ancientTexts-net** [comment]
@elsylambert 

@elsylambert 

- https://develop--tc-create-app.netlify.app/
- v1.3.0-rc.3 build 69-5fefc88

**Comment by elsylambert** [comment]
Looks good in translationCore Create - v1.3.0-rc.4 build 70-7dfd6f5. Ids for a new row starts with a lowercase letter.

**Comment by elsylambert** [comment]
![Screen Shot 2022-01-04 at 12.32.28 AM.png](https://images.zenhubusercontent.com/5d02efb2f425650e637abe10/e8dc9e5d-7608-4ac3-af9e-517a574c34ec)

**Comment by mandolyte** [comment]
@ancientTexts-net Question: I see that tc-create is using datatable-translatable at v0.11.14. But the master branch of datatable-translatable has version in package.json at v0.11.13-rc.4.

Can I be sure that what is in master is what is on NPM? Thanks.

**Comment by mandolyte** [comment]
Note: the code below was used to address the problem. It is not possible for this code to return anything other than a to z as the first character of the id.

@elsylambert can you provide details on how the test works that revealed this?

```js
// ids must begin with a letter
const randomId = ({ length }) => {
  // get the initial letter first
  const letters = ["a", "b", "c", "d", "e", "f", "g",
    "h", "i", "j", "k", "l", "m", "n", "o", "p", "q",
    "r", "s", "t", "u", "v", "w", "x", "y", "z"
  ];
  const random = Math.floor(Math.random() * letters.length);
  const number = Math.random(); // 0.9394456857981651

  // number.toString(36); // '0.xtis06h6'
  if (length > 9) {
    length = 9;
  }

  const id = letters[random] + number.toString(36).substr(2, length-1); // 'xtis06h6'
  return id;
};
```

**Comment by elsylambert** [comment]
@mandolyte As I mentioned that I found this error shown by CV only in Exodus, I realized that the master branch has the ids that starts with a number instead of a lowercase letter. This has never been changed so the new user branch always copies these IDs. 

The code change with respect to this issue is for the new IDs generated with lowercase letter, which works fine.
